draft-ietf-jsonpath-base.md

```diff
@@ -480,32 +519,29 @@ operations which might be considered erroneous, such as indexing beyond the
 end of an array,
 simply result in fewer nodes being selected.
 
-But a selector doesn't just act on a single node: each selector acts on a
-list of nodes and produces a list of nodes, as follows.
-
-After the root selector, the remainder of the JSONPath is processed by passing
-lists of nodes from one selector to the next ending up with a list of nodes
-which is the result of
-applying the JSONPath to the input JSON value.
-
-Each selector acts on its input list of nodes as follows.
-For each node in
-the list, the selector selects zero or more nodes, each of which is a descendant
-of the node or the node itself.
-The output list of nodes of a selector is the concatenation of the lists
-of selected nodes for each input node.
-
-A specific, non-normative example will make this clearer.
-Suppose the input
-document is: `{"a":[{"b":0},{"b":1},{"c":2}]}`.
-As we will see later, the JSONPath `$.a[*].b` selects the following list of nodes: `0`, `1`.
+But a selector doesn't just act on a single node: a selector acts on
+each of the nodes in its input nodelist and selects an output
+nodelist for each of them, as follows, which are then combined using
+the combine1 algorithm to form the result nodelist of the selector.
+
+> Discussion: Do we allow selectors that have their own combine1
+> variant, or can we always use the same?
+
+For each node in the list, the selector selects zero or more nodes,
+each of which is a descendant of the node or the node itself.
+
+> Discussion: Is that a common requirement?  Or can a selector also go
+> up, or to the query Argument?
```

Comment:
This touches on the function of `@`, `$` within filter queries, and the idea of scope when `@` is used in nested filter queries.

I think it's better left for another PR when we've explored these ideas and how they relate.

Reply:
A selector can never go up, given the information the current node is holding alone, as it lacks parent node info.

Maintaining parent node info during selection process introduces a new &ndash; hard to overview &ndash; level of complexity and should be avoided.

Reply:
I agree a selector can never go up.